### PR TITLE
Invoice request delete mutation

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
Invoice request delete mutation, unit tests.

**_Notice that this pull request is to be merged into feature branch, not master._**

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

File: tests/api/test_order.py [hunk 1]
```diff
@@ -2912,13 +2927,49 @@ def test_request_invoice_no_permissions(staff_api_client, orders):
...
 
 @mock.patch("saleor.extensions.base_plugin.BasePlugin.invoice_delete")
-def test_delete_invoice(plugin_mock, user_api_client, permission_manage_orders, orders):
+def test_request_delete_invoice(
+    plugin_mock, user_api_client, permission_manage_orders, orders
+):
+    invoice = Invoice.objects.create(order=orders[0])
+    variables = {"id": graphene.Node.to_global_id("Invoice", invoice.pk)}
+    user_api_client.user.user_permissions.add(permission_manage_orders)
+    user_api_client.post_graphql(REQUEST_DELETE_INVOICE_MUTATION, variables)
```
Comment (fowczarek) [inline tests/api/test_order.py]:
```suggestion
    user_api_client.post_graphql(REQUEST_DELETE_INVOICE_MUTATION, variables, permissions=[permission_manage_orders])
```

File: tests/api/test_order.py [hunk 2]
```diff
@@ -2912,13 +2927,49 @@ def test_request_invoice_no_permissions(staff_api_client, orders):
 
 
 @mock.patch("saleor.extensions.base_plugin.BasePlugin.invoice_delete")
-def test_delete_invoice(plugin_mock, user_api_client, permission_manage_orders, orders):
+def test_request_delete_invoice(
+    plugin_mock, user_api_client, permission_manage_orders, orders
```
Comment (fowczarek) [inline tests/api/test_order.py]:
```suggestion
    plugin_mock, staff_api_client, permission_manage_orders, orders
```

File: tests/api/test_order.py [hunk 3]
```diff
@@ -2912,13 +2927,49 @@ def test_request_invoice_no_permissions(staff_api_client, orders):
...
+):
+    variables = {"id": graphene.Node.to_global_id("Invoice", 1337)}
+    user_api_client.user.user_permissions.add(permission_manage_orders)
+    response = user_api_client.post_graphql(REQUEST_DELETE_INVOICE_MUTATION, variables)
+    content = get_graphql_content(response)
+    errors = content["data"]["requestDeleteInvoice"]["invoiceErrors"][0]
+    assert errors["code"] == InvoiceErrorCode.NOT_FOUND.name
+    assert errors["field"] == "id"
+    assert not plugin_mock.called
```
Comment (fowczarek) [inline tests/api/test_order.py]: We should check `assert_called_once_with` instead of `plugin_mock.called`.

Comment (korycins) [reply tests/api/test_order.py]: @fowczarek imo not in this case. as you can see you have `not` for mock. It confirms that we didn't call any plugin method

File: tests/api/test_order.py [hunk 4]
```diff
@@ -2912,13 +2927,49 @@ def test_request_invoice_no_permissions(staff_api_client, orders):
...
+def test_request_delete_invoice(
+    plugin_mock, user_api_client, permission_manage_orders, orders
+):
+    invoice = Invoice.objects.create(order=orders[0])
+    variables = {"id": graphene.Node.to_global_id("Invoice", invoice.pk)}
+    user_api_client.user.user_permissions.add(permission_manage_orders)
+    user_api_client.post_graphql(REQUEST_DELETE_INVOICE_MUTATION, variables)
+    invoice.refresh_from_db()
+    assert invoice.status == InvoiceStatus.PENDING_DELETE
+    assert plugin_mock.called
```
Comment (korycins) [inline tests/api/test_order.py]: As @fowczarek  said below 👇  Here we can use `assert_called_once_with`

**Pull request merged**